src/context.ts

```diff
@@ -67,6 +84,13 @@ async function getBakeArgs(inputs: Inputs, buildxVersion: string): Promise<Array
   await asyncForEach(inputs.set, async set => {
     args.push('--set', set);
   });
+  if (buildDefinition !== undefined && inputs.ghaCache != GhaCache.Off) {
+    const targetsDefinition: BuildDefinitionTargets = JSON.parse(buildDefinition);
+    for (const target in targetsDefinition.target) {
+      args.push('--set', `${target}.cache-from=type=gha,scope=${github.context.ref}-${target}`);
+      args.push('--set', `${target}.cache-to=type=gha,mode=${inputs.ghaCache},scope=${github.context.ref}-${target}`);
```

Comment:
`github.context.ref` is not right as cache will not be used (cache-from) on any other ref like pull request.

I don't think we should also set cache for all targets. In that case you can just use a wildcard:

```yaml
set: |
  *.cache-from=...
  *.cache-to=...
```

More info about `set` syntax: https://docs.docker.com/reference/cli/docker/buildx/bake/#set

Reply:
> In that case you can just use a wildcard:

That will set the same scope for all targets, which means if you have e.g. multiple builds in one compose file, the different targets will end up overwriting just one cache. This is
* [documented](https://docs.docker.com/build/cache/backends/gha/#scope) 
* [explained by your own comment](https://github.com/docker/bake-action/issues/91#issuecomment-1265817118)

This is the core problem which caused this PR to exist.

> `github.context.ref` is not right as cache will not be used (cache-from) on any other ref like pull request.

Ah, I see. What would be a good cache scope then? Could you use `${target}-${branch}-${pr}-${commithash}`  and somehow have it automatically fall back first to `${target}-${branch}-${pr}-`, `${target}-${branch}-`, and `${target}-` in order?
(That's something that should IMO be on the docs, but since I don't know, I can't write them…)